### DISCUSSION TITLE
Background events

### DISCUSSION THREAD

**Discussion by lukas-frey** [vkurko/calendar#300] [Q&A] [answered]
Hi, can someone give me an example of how to work with background events?

When I set `display: 'background'` on my event, it disappears completely. Am I supposed to add it to my DOM manually?

When I return html from my eventContent function, if I have anything else in `display` than `background` it works, but as soon as I set it to `background`, it doesn't render anything at all.

EDIT: so I checked the docs and found the `theme` method and by the looks of it, there should be an `ec-bg-events` container element somewhere and each event there should be wrapped in an `ec-bg-event` element, if I understand it correctly. Which is how I would imagine it to work.

Is there something else required to do to enable them, except setting `display: 'background'` on the event object?

**Comment by vkurko** [comment]
What view are you using?

**Reply by lukas-frey** [reply]
> What view are you using?

I'm using dayGridMonth

**Reply by vkurko** [reply, marked as answer]
`dayGridMonth` currently does not display background events. Sorry this is not shown in the documentation. FullCalendar also does not display all background events in this view, but only those whose dates are indicated without the time part. Perhaps Event Calendar should work the same way.

**Reply by lukas-frey** [reply]
Ah I see, thanks for the info! Would be great if the allDay events would still be displayed there.

I'm not sure whether my use case is even correct for the background events, but I thought about displaying multi-day, all day events such as a `Sprint` in a project-planning tool to be displayed as background events, whereas something like a `Meeting` (typically a few hours, one-day event) would be the regular events.

I think that fits well into the way you just described that it works in Full Calendar and maybe should work in Event Calendar too.

Great work on the package by the way! I created a [wrapper](https://github.com/GuavaCZ/calendar) for Laravel FilamentPHP and have gotten great feedback so far. :)

**Reply by vkurko** [reply]
> I think that fits well into the way you just described that it works in Full Calendar and maybe should work in Event Calendar too.

Yes, I will implement this in one of the future releases.


> I created a [wrapper](https://github.com/GuavaCZ/calendar) for Laravel FilamentPHP and have gotten great feedback so far.

What a great job you have done! Thank you for that. If you need any more help, be sure to contact me!

**Reply by lukas-frey** [reply]
Thanks! I actually might contact you sometime regarding one issue I'm having, if that's okay.

**Reply by vkurko** [reply]
All-day background events should now be displayed in `dayGridMonth` view. Please check Event Calendar v3.3.0.

**Reply by lukas-frey** [reply]
Thanks, works perfectly!
